common/src/main/java/org/broadleafcommerce/common/resource/service/ResourceMinificationServiceImpl.java

```diff
@@ -110,6 +124,18 @@ public Resource minify(Resource originalResource, String filename) {
             return originalResource;
         }
 
+        Integer filenameLength = originalResource.getFilename().length();
+
+        if (filenameLength >= JS_MIN.length()) {
+
+            String lastSevenCharacters = originalResource.getFilename().substring(filenameLength-JS_MIN.length());
+            if (lastSevenCharacters.equals(JS_MIN)) {
+                LOG.info("Minification already have done for this resource: " + filename);
+                return originalResource;
+            }
+
```

Comment:
It's better to have this logic in its own method so that if a client wants to determine what is or what isn't a minified file they can do that easily. Additionally this can all be replaced by

```
if (StringUtils.endsWith(filename, JS_MIN)) {
    // Info log and return resource
}
```
Typically there's an apache commons StringUtils method for most String operations. This goes for Collections, Integers, Longs, Arrays, etc. It's a good habit to check if there's an apache commons method you can use instead so that we don't have to reinvent the wheel. If you have a feeling that there's probably a library that does something but you can't think of what the static class would be just ask in the Slack channel "developers" and someone will be able to help.